packages/api-client/tests/integration.spec.ts

```diff
@@ -0,0 +1,184 @@
+  PROJECT_UPDATED = 'PROJECT_UPDATED',
+  PROJECT_DELETED = 'PROJECT_DELETED',
+  SECRET_UPDATED = 'SECRET_UPDATED',
+  SECRET_DELETED = 'SECRET_DELETED',
+  SECRET_ADDED = 'SECRET_ADDED',
+  VARIABLE_UPDATED = 'VARIABLE_UPDATED',
+  VARIABLE_DELETED = 'VARIABLE_DELETED',
+  VARIABLE_ADDED = 'VARIABLE_ADDED',
+  ENVIRONMENT_UPDATED = 'ENVIRONMENT_UPDATED',
+  ENVIRONMENT_DELETED = 'ENVIRONMENT_DELETED',
+  ENVIRONMENT_ADDED = 'ENVIRONMENT_ADDED',
+  INTEGRATION_ADDED = 'INTEGRATION_ADDED',
+  INTEGRATION_UPDATED = 'INTEGRATION_UPDATED',
+  INTEGRATION_DELETED = 'INTEGRATION_DELETED'
+}
+
+describe('Get Environments Tests', () => {
+  const backendUrl = process.env.BACKEND_URL as string
+
+  const client = new APIClient(backendUrl)
+  const integrationController = new IntegrationController(backendUrl)
+  const email = 'johndoe@example.com'
+  let projectId: string | undefined
+  let workspaceId: string
+  let environment: any
+  let integrationId: string
+
+  beforeAll(async () => {
+    //Create the user's workspace
+    const workspaceResponse = (await (
+      await client.post(
+        '/api/workspace',
+        {
+          name: 'Integration Workspace'
+        },
+        {
+          'x-e2e-user-email': email
+        }
+      )
+    ).json()) as any
+
+    workspaceId = workspaceResponse.id
+
+    // Create a project
+    const projectResponse = (await (
+      await client.post(
+        `/api/project/${workspaceId}`,
+        {
+          name: 'Project',
+          storePrivateKey: true
+        },
+        {
+          'x-e2e-user-email': email
+        }
+      )
+    ).json()) as any
+
+    projectId = projectResponse.id
+
+    const createEnvironmentResponse = (await (
+      await client.post(
+        `/api/environment/${projectId}`,
+        {
+          name: 'Dev'
+        },
+        {
+          'x-e2e-user-email': email
+        }
+      )
+    ).json()) as any
+
+    environment = createEnvironmentResponse
+  })
+
+  afterAll(async () => {
+    // Delete the workspace
+    await client.delete(`/api/workspace/${workspaceId}`, {
+      'x-e2e-user-email': email
+    })
+  })
+
+  it('should create a integration', async () => {
+    const integration = await integrationController.createIntegration(
+      {
+        workspaceId,
+        projectId,
+        name: 'Discord second',
+        type: IntegrationType.DISCORD,
+        notifyOn: [EventType.PROJECT_CREATED],
+        metadata: {
+          webhookUrl: '{{vault:WEBHOOK_URL}}'
+        },
+        environmentId: environment.id
+      },
+      {
+        'x-e2e-user-email': email
+      }
+    )
+    expect(integration.data?.name).toBe('Discord second')
+    expect(integration.data?.projectId).toBe(projectId)
+    expect(integration.data?.environmentId).toBe(environment.id)
+    expect(integration.data?.workspaceId).toBe(workspaceId)
+    expect(integration.data?.type).toBe('DISCORD')
+    integrationId = integration.data?.id as string
```

Comment:
Refrain from using this integration as a global one. Create a dummy integration in `beforeEach` and `afterEach` blocks. Refer to the other controller tests to understand how its done.